errors.go

```diff
@@ -106,6 +106,23 @@ func New(message string) error {
 	}
 }
 
+// ExistStack return an error already with stack
+// ExistStack will check all parent error
+func ExistStack(err error) bool {
+	type stackTracer interface {
+		StackTrace() StackTrace
+	}
+	if _, ok := err.(stackTracer); ok {
+		return true
+	}
+	type causer interface {
+		Cause() error
+	}
+	if _, ok := err.(causer); !ok {
+		return false
+	}
+	return ExistStack(err.Cause())
```

Comment:
🤔 Not sure if the Go compiler is smart enough to rework end-recursion into the more efficient for-loop.